src/main/java/org/testng/IInjectorFactory.java

```diff
@@ -11,11 +12,12 @@
 public interface IInjectorFactory {
 
   /**
+   * @param parent - Parent {@link com.google.inject.Injector} instance that was built with parent injector
    * @param stage - A {@link Stage} object that defines the appropriate stage
    * @param modules - An array of {@link Module}
-   * @return - An {@link com.google.inject.Inject} instance that can be used to perform dependency
+   * @return - An {@link com.google.inject.Injector} instance that can be used to perform dependency
    * injection.
    */
-  Injector getInjector(Stage stage, Module... modules);
+  Injector getInjector(@Nullable Injector parent, Stage stage, Module... modules);
```

Comment:
We can't change the public API easily.
Please deprecate the previous method first and add a default behavior for the new one.

Reply:
Done. Note that it cannot be marked as @FunctionalInterface anymore as it contains more methods now...

Reply:
Right. If you prefer, a new interface could be used instead.

Reply:
Yeah, was wondering about going into that direction as removing @FunctionalInterface is somewhat invasive although TBH it is not used that way in the existing code hence there is no big risk unless someone uses it that way in their code (can't think why would they ;))... anyway I have no strong opinion on any solution and will follow the lead here ;) So what is your take on that?

Reply:
I don't have any strong opinion too.

@krmahadevan WDYT?

Reply:
@juherr - I believe that we will ONLY be temporarily revoking the `FunctionalInterface` stature of this interface till we get through with this deprecation strategy and remove off the original method. So it should be fine if we change the stature of this interface to a regular interface. I am fine with this change.